boltdesignsystem/bolt
# Feature/ie11 deprecation

## Jira

- https://pegadigitalit.atlassian.net/browse/DS-202
- http://vjira2:8080/browse/BDS-2026

## Summary

Remove IE11 support from build tools, config files, and theming system.

## Details

### Build tools
- Remove IE11 from `@bolt/browserslist-config`
- Replace legacy/modern config with single config in `@bolt/babel-preset-bolt`
- Remove `esModules` and `compat` flags
- Remove "modern" from filenames

### Theming
- Remove `@bolt/postcss-themify` and `@bolt/components-critical-css-vars`
  - Remove option to pass `map()` to `bolt-theme()` in [`_tools.themes.scss`](https://github.com/boltdesignsystem/bolt/pull/1801/files#diff-7e1807eb85ca58670e22453293d92266)
  - Replace uses of `map()` with a single CSS variable, make necessary component updates (detailed below)

#### Component updates
- Accordion: update snapshot to reflect CSS formatting changes, no visual difference ([see file](https://github.com/boltdesignsystem/bolt/pull/1801/files#diff-dcb659d6f2bb7f30d53ee49ed90198db)).
- Card Replacement: replace background-color `map()` with single CSS variable; results in background-color that is semi-transparent, not solid on light theme ([see file](https://github.com/boltdesignsystem/bolt/pull/1801/files#diff-a7d393cabdb2721b66e1fb47dd50139b)).
- Nav Priority: Replace background-color `map()` with single CSS variable; Add styles to approximate what was previously done [via `map()`](https://github.com/boltdesignsystem/bolt/pull/1801/files#diff-a493a92d67db76d80f7c21d0f18d147aL142), which is to make the menu lighter on dark themes. The result is slightly lighter than it previously was on `xdark`, slightly darker than it was on `dark`.

### Follow up
More work queued up once this is approved:
  - Removed Polyfills
  - Updated PL and Integrations

## How to test

- Review changes
- Tests are passing
- Review [Card Replacement demo](https://feature-ie11-deprecation.boltdesignsystem.com/pattern-lab/patterns/40-components-card-replacement-20-card-replacement-theme-variations/40-components-card-replacement-20-card-replacement-theme-variations.html) and verify light theme looks ok
- Review [Priority Nav demo](https://feature-ie11-deprecation.boltdesignsystem.com/pattern-lab/patterns/40-components-nav-priority-10-nav-priority--multiple-sticky-navbars/40-components-nav-priority-10-nav-priority--multiple-sticky-navbars.html) and verify dropdown menu (seen on smaller screens) looks ok
- Run the branch locally and do a spot check of other components (be sure to run `yarn`)
- ~For @remydenton, pull into Bolt Build Release and see how it runs ([created ticket](http://vjira2:8080/browse/BDS-2090))~ There is a [follow-up ticket](https://pegadigitalit.atlassian.net/browse/DS-235) to remove unused polyfills. You might want to wait until this second PR is merged before testing in Drupal.
